ob1k-cache/src/main/java/com/outbrain/ob1k/cache/metrics/AsyncBulkGetMetrics.java

```diff
@@ -0,0 +1,42 @@
+package com.outbrain.ob1k.cache.metrics;
+
+import com.outbrain.ob1k.concurrent.Try;
+import com.outbrain.swinfra.metrics.api.Counter;
+import com.outbrain.swinfra.metrics.api.MetricFactory;
+import com.outbrain.swinfra.metrics.api.Timer;
+import java.util.Map;
+
+/**
+ * Holds a set of metrics used for measuring get operations.
+ *
+ * @author hunchback
+ */
+public class AsyncBulkGetMetrics extends AsyncOperationMetrics {
+  private final Counter hits;
+  private final Counter total;
+
+  public AsyncBulkGetMetrics(final MetricFactory metricFactory, final String component) {
+    super(metricFactory, component);
+    hits = metricFactory.createCounter(component, "cacheHits");
+    total = metricFactory.createCounter(component, "cacheTotal");
+  }
+
+  public void increaseTotal(final Iterable keys) {
+    for (final Object key : keys) {
+      total.inc();
+    }
+  }
+
+  @Override
+  public void finished(final Try res, final Timer.Context timer) {
+    super.finished(res, timer);
+
+    if (res.isSuccess()) {
+      if (res.getValue() != null) {
```

Comment:
In LoadingCacheDelegate which decorates a cache the value is checked for "null".
In LocalAsyncCache there are explicit "res = fromValue(null)" in cases of getting a "null" from the lower level cache implementations it depend on.

So from this I conclude that I should check for "null".